app/lib/handlers/pen.dart

```diff
@@ -162,13 +140,240 @@ class PenHandler extends Handler<PenTool> with ColoredHandler {
+
+            // Convert coordinates from the document coordinate system to the view coordinate system
+            Offset firstPositionInView =
+                transform.localToGlobal(firstPosition.toOffset());
+            Offset secondPositionInView =
+                transform.localToGlobal(secondPosition.toOffset());
+
+            // Create new ShapeElement
+            PadElement shapeElement = PadElement.shape(
+              firstPosition: firstPositionInView.toPoint(),
+              secondPosition: secondPositionInView.toPoint(),
+              property: ShapeProperty(
+                  shape: const CircleShape(),
+                  color: data.property.color,
+                  strokeWidth: data.property.strokeWidth),
+            );
+
+            // show SnackBar with recognized shape
+            ScaffoldMessenger.of(context.buildContext).showSnackBar(
+              SnackBar(
+                width: MediaQuery.of(context.buildContext).size.width * 0.1,
+                behavior: SnackBarBehavior.floating,
+                content: Text(
+                  textAlign: TextAlign.center,
+                  AppLocalizations.of(context.buildContext).circle,
+                ),
+                duration: const Duration(milliseconds: 300),
+              ),
+            );
+            // Add element on document
+            context.getDocumentBloc().add(ElementsCreated([shapeElement]));
+
+            _timer?.cancel();
+            elements.clear();
+            context.refresh();
+          } else if (points.length > 700) {
+            return;
+          }
+          break;
+
+        case DefaultUnistrokeNames.rectangle:
+          if (element != null && points.length < 700) {
+            double minX = points.map((p) => p.dx).reduce(min);
+            double maxX = points.map((p) => p.dx).reduce(max);
+            double minY = points.map((p) => p.dy).reduce(min);
+            double maxY = points.map((p) => p.dy).reduce(max);
+
+            Point<double> firstPosition = Point(minX, minY);
+            Point<double> secondPosition = Point(maxX, maxY);
+
+            // Convert coordinates from the document coordinate system to the view coordinate system
+            Offset firstPositionInView =
+                transform.localToGlobal(firstPosition.toOffset());
+            Offset secondPositionInView =
+                transform.localToGlobal(secondPosition.toOffset());
+
+            // Create new ShapeElement
+            PadElement shapeElement = PadElement.shape(
+              firstPosition: firstPositionInView.toPoint(),
+              secondPosition: secondPositionInView.toPoint(),
+              property: ShapeProperty(
+                  shape: const RectangleShape(),
+                  color: data.property.color,
+                  strokeWidth: data.property.strokeWidth),
+            );
+
+            // show SnackBar with recognized shape
+            ScaffoldMessenger.of(context.buildContext).showSnackBar(
```

Comment:
can you move this logic out of the switch so we don't duplicate code?
We could save the content string of the recognized shape inside a variable